server/protocols/jmap-draft/src/main/java/org/apache/james/jmap/http/UploadRoutes.java

```diff
@@ -115,7 +112,7 @@ private UploadRoutes(MetricFactory metricFactory, Authenticator authenticator, A
     }
 
     private Mono<Void> post(HttpServerRequest request, HttpServerResponse response, String contentType, MailboxSession session) {
-        InputStream content = ReactorUtils.toInputStream(request.receive().asByteBuffer());
+        InputStream content = ReactorUtils.toInputStream(request.receive().asByteBuffer().subscribeOn(Schedulers.elastic()));
```

Comment:
Because toInputStream calls `Flux::toStream` whose javadoc docments blocking on each entry.

If we want to do anything here we need type changes (`Mono<InputStream>` ?)

Anyway such a fix is imo out of the scope of this work.

Should I open a ticket about it?